inc/Engine/Media/Embeds/EmbedsSubscriber.php

```diff
@@ -0,0 +1,237 @@
+	 */
+	public function disable_embeds_tiny_mce_plugin( $plugins ) {
+		if ( ! $this->can_embed() ) {
+			return $plugins;
+		}
+
+		return array_diff( $plugins, [ 'wpembed' ] );
+	}
+
+	/**
+	 * Remove all rewrite rules related to embeds.
+	 *
+	 * @since  3.7 Moved to new architecture.
+	 * @since  2.10
+	 *
+	 * @param array $rules WordPress rewrite rules.
+	 *
+	 * @return array Rewrite rules without embeds rules.
+	 */
+	public function disable_embeds_rewrites( $rules ) {
+		if ( empty( $rules ) || ! $this->can_embed() ) {
+			return $rules;
+		}
+
+		foreach ( $rules as $rule => $rewrite ) {
+			if ( false !== strpos( $rewrite, 'embed=true' ) ) {
+				unset( $rules[ $rule ] );
+			}
+		}
+
+		return $rules;
+	}
+
+	/**
+	 * Remove the oembed/1.0/embed REST route.
+	 *
+	 * @since 3.7 Moved to new architecture.
+	 * @since 3.3.3
+	 *
+	 * @param array $endpoints Registered REST API endpoints.
+	 *
+	 * @return array Filtered REST API endpoints.
+	 */
+	public function disable_embeds_remove_embed_endpoint( $endpoints ) {
+		if ( ! $this->can_embed() ) {
+			return $endpoints;
+		}
+
+		unset( $endpoints['/oembed/1.0/embed'] );
+
+		return $endpoints;
+	}
+
+	/**
+	 * Disables sending internal oEmbed response data in proxy endpoint.
+	 *
+	 * @since 3.7 Moved to new architecture.
+	 * @since 3.3.3
+	 *
+	 * @param array $data The response data.
+	 *
+	 * @return array|false Response data or false if in a REST API context.
+	 */
+	public function disable_embeds_filter_oembed_response_data( $data ) {
+		if ( defined( 'REST_REQUEST' ) && REST_REQUEST ) {
+			return false;
+		}
+
+		if ( ! $this->can_embed() ) {
+			return false;
+		}
+
+		return $data;
+	}
+
+	/**
+	 * Enqueue JavaScript for the block editor.
+	 *
+	 * @since 3.7 Moved to new architecture.
+	 * @since 3.3.3
+	 *
+	 * This is used to unregister the `core-embed/wordpress` block type.
+	 */
+	public function disable_embeds_enqueue_block_editor_assets() {
+		if ( ! $this->can_embed() ) {
+			return;
+		}
+
+		wp_enqueue_script(
+			'rocket-disable-embeds',
+			WP_ROCKET_ASSETS_JS_URL . 'editor/editor.js',
+			[
+				'wp-edit-post',
+				'wp-editor',
+				'wp-dom',
+			],
+			WP_ROCKET_VERSION,
+			true
+		);
+	}
+
+	/**
+	 * Remove wp-embed dependency of core packages.
+	 *
+	 * @since 3.7 Moved to new architecture.
+	 * @since 3.3.3
+	 *
+	 * @param \WP_Scripts $scripts WP_Scripts instance, passed by reference.
+	 */
+	public function disable_embeds_remove_script_dependencies( $scripts ) {
+		if ( ! $this->can_embed() ) {
+			return;
+		}
+
+		if ( ! empty( $scripts->registered['wp-edit-post'] ) ) {
+			$scripts->registered['wp-edit-post']->deps = array_diff(
+				$scripts->registered['wp-edit-post']->deps,
+				[ 'wp-embed' ]
+			);
+		}
+	}
+
+	/**
+	 * Check for embeds enabled.
+	 *
+	 * @since 3.7
+	 *
+	 * @return bool
+	 */
+	private function can_embed() {
+		if ( rocket_bypass() || ! $this->options->get( 'embeds' ) ) {
+			return false;
+		}
+
+		return true;
```

Comment:
We can simplify this as a one liner

```suggestion
		return ( ! rocket_bypass() || (bool) $this->options->get( 'embeds', 0 ) );
```